compiler/aarch64/codegen/GenerateInstructions.cpp

```diff
@@ -250,3 +250,31 @@ TR::Instruction *generateLogicalShiftLeftImmInstruction(TR::CodeGenerator *cg, T
       return new (cg->trHeapMemory()) TR::ARM64Trg1Src1ImmInstruction(op, node, treg, sreg, imm, preced, cg);
    return new (cg->trHeapMemory()) TR::ARM64Trg1Src1ImmInstruction(op, node, treg, sreg, imm, cg);
    }
+
+TR::Instruction *generateCompareImmInstruction(TR::CodeGenerator *cg, TR::Node *node,
+   TR::Register *sreg, int32_t imm, TR::Instruction *preced)
+   {
+   /* Alias of SUBS instruction */
+
+   bool is64bit = node->getDataType().isInt64();
+   TR::InstOpCode::Mnemonic op = is64bit ? TR::InstOpCode::Mnemonic::subsimmx : TR::InstOpCode::Mnemonic::subsimmw;
+   TR::RealRegister *zeroReg = cg->machine()->getARM64RealRegister(TR::RealRegister::xzr);
+
+   if (preced)
+      return new (cg->trHeapMemory()) TR::ARM64Trg1Src1ImmInstruction(op, node, zeroReg, sreg, imm, preced, cg);
+   return new (cg->trHeapMemory()) TR::ARM64Trg1Src1ImmInstruction(op, node, zeroReg, sreg, imm, cg);
+   }
+
+TR::Instruction *generateTestImmInstruction(TR::CodeGenerator *cg, TR::Node *node,
+   TR::Register *sreg, int32_t imm, TR::Instruction *preced)
+   {
+   /* Alias of ANDS instruction */
+
+   bool is64bit = node->getDataType().isInt64();
+   TR::InstOpCode::Mnemonic op = is64bit ? TR::InstOpCode::Mnemonic::andsimmx : TR::InstOpCode::Mnemonic::andsimmw;
+   TR::RealRegister *zeroReg = cg->machine()->getARM64RealRegister(TR::RealRegister::xzr);
+
+   if (preced)
+      return new (cg->trHeapMemory()) TR::ARM64Trg1Src1ImmInstruction(op, node, zeroReg, sreg, imm, preced, cg);
+   return new (cg->trHeapMemory()) TR::ARM64Trg1Src1ImmInstruction(op, node, zeroReg, sreg, imm, cg);
+   }
```

Comment:
Other than the opcodes in play, these two functions are identical.  You can save some .so size by refactoring the bulk of this logic into a common function used by both functions that you simply pass the chosen opcode to.

Reply:
Changed.